ServerSide/yoty/Core/Managers/Bids/BidsManager.cs

```diff
@@ -232,12 +235,110 @@ public async Task<Response<List<BuyerDTO>>> GetBidBuyers(string bidId)
             List<BuyerDTO> buyers = new List<BuyerDTO>();
             bid_ent.CurrentParticipancies.ForEach(p => buyers.Add(_mapper.Map<BuyerDTO>(p.Buyer)));
             return new Response<List<BuyerDTO>>() { DTOObject = buyers, IsOperationSucceeded = true, SuccessOrFailureMessage = this.getSuccessMessage() };
+        }
+
+        public async Task<Response<List<BidDTO>>> GetBids(BidsQueryOptions bidsFilters)
+        {
+
+            // firts category and sub category 
+            // then 
+            if (bidsFilters.Category == null && bidsFilters.Search == null)
+            {
+                return this.GetDefaultHomePageBids(bidsFilters.Page);
+            }
+            else if (!ValidateBidsFilters(bidsFilters, out string validationErorrString))
+            {
+                return new Response<List<BidDTO>>() { DTOObject = null, IsOperationSucceeded = false, SuccessOrFailureMessage = validationErorrString };
+            }
+            else // TODO : extract to sub methods
+            {
+                IEnumerable<BidEntity> filteredBids = _context.Bids
+                    // filter by Categories
+                    .Where(bid => bid.Category == null ?
+                                true :
+                                bid.Category.Equals(bidsFilters.Category) &&
+                                (bidsFilters.SubCategory == null) ?
+                                    true :
+                                    bid.SubCategory != null && bid.SubCategory.Equals(bidsFilters.Category))
+                    //filter by prices
+                    .Where(bid => bidsFilters.MaxPrice < Int32.MaxValue ? bid.MaxPrice < bidsFilters.MaxPrice : true &&
+                            bidsFilters.MinPrice > 0 ? bid.MaxPrice > bidsFilters.MinPrice : true)
+                    // filter by query string
+                    .Where(bid => bid.Product.Name.Contains(bidsFilters.Search) || bid.Product.Description.Contains(bidsFilters.Search));
```

Comment:
please add .Include(bid => bid.Product)

if you remember we talked about it not being efficient to pull all participants when querying for a bid. we were right, that is why context acts in a lazy manner, you must tell it to include other referenced entities (in this case Product) otherwise it will only populate the fields that are columns in the Bids table and others will be null (that was the issue I had)